Generalize ApprovalDenied/RejectionDenied to ParticipationDenied (#115)
upb-uc4/api: hlf/chaincode/errors.md

reason:
- a textual representation of why the parameter is invalid

# Specific Errors

> In the following, the specific error that can be returned by every transaction not part of the [OperationContract](./contracts/operation.md#OperationApi)  due to our operation workflow.

Author: ```suggestion
In the following, the specific error that can be returned by every transaction due to our operation workflow.

```

- [GenericError](#GenericError) 
  ```json
  {
    "type": "HLInsufficientApprovals",
    "title": "The approvals present on the ledger do not suffice to execute this transaction"
  }
  ```
  - Description: This error is returned, if the required approvals are not present.

- [GenericError](#GenericError) 
  ```json
  {
    "type": "HLParticipationImpossible",
    "title": "The operation is not in pending state"
  }
  ```
  - Description: This error is returned, if the operation for this transaction is not pending.

- [GenericError](#GenericError) 
  ```json
  {

>     "type": "HLParticipationDenied",
>     "title": "You are not allowed to participate in the given operation"

Author: ```suggestion
    "type": "HLAccessDenied",
    "title": "You are not allowed to execute in the given transaction"
```

reason:
- a textual representation of why the parameter is invalid

# Specific Errors

## <a id="AllTransactions" />All Transactions

In the following, the specific errors that can be returned by every transaction due to our operation workflow are defined.

- [GenericError](#GenericError) 
  ```json
  {
    "type": "HLInsufficientApprovals",
    "title": "The approvals present on the ledger do not suffice to execute this transaction"
  }
  ```
  - Description: This error is returned, if the required approvals are not present.

- [GenericError](#GenericError) 
  ```json
  {
    "type": "HLAccessDenied",
    "title": "You are not allowed to execute the given transaction"
  }
  ```
    - Description: This error is returned, if the user trying to execute the transaction is not allowed to participate in the operation.

## <a id="AllOperations" />All Operations

In the following, the specific errors that can additionally be returned by every operation are defined.

- [GenericError](#GenericError) 
  ```json
  {

>     "type": "HLParticipationImpossible",

Reviewer: ```suggestion
    "type": "HLExecutionImpossible",
```
